src/frontend/lib/mapeo-schema.d.ts

```diff
@@ -0,0 +1,189 @@
+// Refer to https://github.com/digidem/mapeo-schema/blob/master/docs/README.md
+declare module "mapeo-schema" {
+  // TODO: do we need this?
+  interface Common {
+    created_at: string;
+    deviceId?: string;
+    id: string;
+    links?: string[];
+    schemaVersion: number;
+    timestamp?: string;
+    userId?: string;
+    type: string;
+  }
+
+  export interface Position {
+    coords?: {
+      accuracy?: number;
+      altitude?: number;
+      heading?: number;
+      latitude?: number;
+      longitude?: number;
+      speed?: number;
+    };
+    mocked?: boolean;
+  }
+
+  export interface Observation {
+    attachments?: {
+      id: string;
+      type?: string;
+    }[];
+    created_at: string;
+    deviceId?: string;
+    id: string;
+    lat?: number | null;
+    links?: string[];
+    lon?: number | null;
+    metadata?: {
+      lastSavedPosition?: Position;
+      manualLocation?: boolean;
+      position?: Position;
+      positionProvider?: {
+        gpsAvailable?: boolean;
+        locationServicesEnabled?: boolean;
+        networkAvailable?: boolean;
+        passiveAvailable?: boolean;
+      };
+    };
+    refs?: {
+      id: string;
+    }[];
+    schemaVersion: number;
+    tags?: {
+      // TODO
+      [key: string]: any;
+    };
+    timestamp?: string;
+    type: "observation";
+    userId?: string;
+    version: string;
+  }
+
+  export interface Preset {
+    addTags?: {
+      [key: string]: any;
+    };
+    additionalFields?: string[];
+    fields?: string[];
+    geometry: ("point" | "vertex" | "line" | "area" | "relation")[];
+    icon?: string;
+    id: string;
+    name: string;
+    removeTags?: {
+      [key: string]: any;
+    };
+    schemaVersion: number;
+    sort?: number;
+    tags: {
+      [key: string]: any;
+    };
+    terms?: string[];
+  }
+
+  // TODO
+  type MbFilter = (string | number | boolean | MbFilter)[];
+
+  export interface Filter {
```

Comment:
Not yet implemented.

Reply:
can remove to reduce noise if desired 👍

Reply:
Yes let's remove for now. Maybe these types can be published with the `mapeo-schema` repo in the future.

Reply:
Cool. yeah ideally we'd just add this type def file to `mapeo-schema` instead of creating it here